docs/pages/upgrading/self-hosted-kubernetes.mdx

```diff
@@ -0,0 +1,153 @@
+---
+title: Upgrade Self-Hosted Teleport Clusters on Kubernetes
+description: Provides instructions for upgrading self-hosted Teleport clusters that run on Kubernetes. 
+---
+
+This guide explains how to upgrade self-hosted Teleport clusters running on
+Kubernetes.
+
+## Prerequisites
+
+- Familiarity with the [Upgrading Compatibility Overview](./overview.mdx) guide,
+  which describes the sequence in which to upgrade components of your cluster.
+- A self-hosted Teleport cluster in which the Auth Service and Proxy Service run
+  on Kubernetes. This guide assumes that you have deployed the Teleport cluster
+  using the `teleport-cluster` Helm chart.
+
+Teleport supports automatic agent updates for the `teleport-kube-agent` Helm
+chart. The [Automatic Update Architecture
+](../architecture/agent-update-management.mdx) guide describes how agent
+updating works. Automatic agent upgrades require:
+
+- A Teleport cluster running version 13.0.0 or above.
+- At least one Teleport Enterprise agent deployed via the `teleport-kube-agent`
+  Helm chart.
+- A **version server** in your infrastructure that agents can query in order to
+  upgrade themselves. Read [Set up Automatic Upgrade
+  Infrastructure](./self-hosted-automatic-agent-updates.mdx) for how to deploy a
+  version server.
+
+This guide assumes that you have configured the `teleport-cluster` Helm chart
+with a values file called `values.yaml`, and that your `teleport-cluster`
+release is called `teleport-cluster`.
+
+## Step 1/3. Shrink the Auth Service pool
+
+You must reduce the number of Auth Service instances to one in order to ensure a
+consistent cluster state during the upgrade.
+
+Ensure that your `teleport-cluster` values file includes the followinmg
+configuration:
+
+```yaml
+auth:
+  highAvailability:
+    replicaCount: 1
+```
+
+Once you have completed this guide and upgraded the cluster, you can configure
+your cluster for high availability again.
+
+## Step 2/3. Upgrade the Auth Service and Proxy Service
+
+Run the following commands to upgrade Auth Service and Proxy Service instances
+running on Kubernetes.
+
+1. Update the Teleport Helm chart repository so you can install the latest
+   version of the `teleport-cluster` chart:
+
+    (!docs/pages//kubernetes-access/helm/includes/helm-repo-add.mdx!)
+
+1. Upgrade the Helm release:
+
+    ```code
+    $ helm upgrade teleport-cluster teleport/teleport-cluster --values=values.yaml
+    ```
+
+The `teleport-cluster` Helm chart automatically waits for the previous version
+of the Proxy Service to stop responding to requests before running a new version
+of the Auth Service.
+
+
+## Step 3/3. Enroll agents in automatic upgrades
```

Comment:
I don't think setting up automatic updates should be the next step in self-hosted setups because:
- we don't support automatic updates for Teleport OSS (because of some sad packaging issue, there's no technical or business limitation)
- most self-hosted users don't have automatic updates version servers. Teleport is not yet hosting its own version server (I hope it will become one day, so we can use automatic updates everywhere).
- there are still a lot of setups not supported by automatic upgrades